zebra-state/src/service/finalized_state/zebra_db/transparent.rs

```diff
@@ -160,6 +161,47 @@ impl ZebraDb {
 
         addr_unspent_outputs
     }
+
+    /// Returns the locations of any transactions sent to a [`transparent::Address`],
+    /// if they are in the finalized state.
+    #[allow(dead_code)]
+    pub fn address_transaction_locations(
+        &self,
+        address_location: AddressLocation,
+    ) -> BTreeSet<AddressTransaction> {
+        let tx_loc_by_transparent_addr_loc =
+            self.db.cf_handle("tx_loc_by_transparent_addr_loc").unwrap();
+
+        // Manually fetch the entire addresses' transaction locations
+        let mut addr_transactions = BTreeSet::new();
+
+        // An invalid key representing the minimum possible transaction
+        let mut transaction_location = AddressTransaction::address_iterator_start(address_location);
+
+        loop {
+            // A valid key representing an entry for this address or the next
+            transaction_location = match self
+                .db
+                .zs_next_key_value_from(&tx_loc_by_transparent_addr_loc, &transaction_location)
+            {
+                Some((unspent_output, ())) => unspent_output,
+                // We're finished with the final address in the column family
+                None => break,
+            };
+
+            // We found the next address, so we're finished with this address
+            if transaction_location.address_location() != address_location {
+                break;
+            }
+
+            addr_transactions.insert(transaction_location);
+
+            // A potentially invalid key representing the next possible output
+            transaction_location.address_iterator_next();
+        }
+
+        addr_transactions
+    }
 }
```

Comment:
For the `get_address_tx_ids` RPC, we need to return the transaction hashes in chain order.

Can you add a method something like this?
```rust
    /// Returns the [`transaction::Hash`]es that created or spent outputs for a [`transparent::Address`],
    /// in chain order, if they are in the finalized state.
    #[allow(dead_code)]
    pub fn address_tx_ids(
        &self,
        address: &transparent::Address,
    ) -> BTreeMap<TransactionLocation, transaction::Hash> { ... }
```

Reply:
Done in bcd8d0b8c